streams/src/test/java/org/apache/kafka/streams/kstream/internals/WindowedStreamPartitionerTest.java

```diff
@@ -138,10 +138,88 @@ public void testWindowedDeserializerNoArgConstructors() {
         props.remove("value.deserializer.inner.class");
         WindowedDeserializer<?> windowedDeserializer1 = new WindowedDeserializer<>();
         windowedDeserializer1.configure(props, false);
-        Deserializer<?> inner1 = windowedDeserializer1.innerDeserializer();
+        final Deserializer<?> inner1 = windowedDeserializer1.innerDeserializer();
         assertNotNull("Inner deserializer should be not null", inner1);
         assertTrue("Inner deserializer type should be ByteArrayDeserializer", inner1 instanceof ByteArrayDeserializer);
         windowedDeserializer.close();
         windowedDeserializer1.close();
     }
+    
+    @Test
+    public void testWindowSerializeConfig() {
+        final Map<String, String> props = new HashMap<>();
+        // test key[value].serializer.inner.class takes precedence over serializer.inner.class
+        final WindowedDeserializer<StringSerializer> windowedDeserializer = new WindowedDeserializer<>();
+        props.put("key.deserializer.inner.class", "org.apache.kafka.common.serialization.StringDeserializer");
+        props.put("deserializer.inner.class", "org.apache.kafka.common.serialization.StringDeserializer");
+        windowedDeserializer.configure(props, true);
+        final Deserializer<?> inner = windowedDeserializer.innerDeserializer();
+        assertTrue("Inner deserializer type should be StringDeserializer", inner instanceof StringDeserializer);
+    }
+    
+    @Test
+    public void testWindowSerializeWithWindowSize() {
+        final Map<String, String> props = new HashMap<>();
+        // test key[value].serializer.inner.class takes precedence over serializer.inner.class
```

Comment:
this is tested in first test method already, and can be removed. (not just the comment, but also the corresponding code.)